db/articles.js

```diff
@@ -0,0 +1,75 @@
+//jshint esversion:6
+class Articles {
+  constructor(){
+    this._collection = [];
+  }
+
+  getArticles (){
+    return this._collection;
+  }
+  post (article){
+    let isRepost = articleIsRepost(article.title, this._collection);
+    if(isRepost === true){
+      const newProduct = {
+        title : article.title,
+        body : article.body,
+        author : article.author,
+        URL : encodeURI(article.title),
+      };
+    this._collection.push(newProduct);
+    return newProduct;
+    }
+  }
+  putArticle(url, data){
+    for(let i = 0; i < this._collection.length; i++){
+      if(url === this._collection[i].URL) {
```

Comment:
I would recommend pulling this work out of the for loop and doing it below. The for loop should just grab the particular article to be edited if any at all.